tools/hardware/utils/memory.py

```diff
@@ -26,6 +26,36 @@ def visitor(node: WrappedNode):
     return regions
 
 
+def merge_memory_regions(regions: Set[Region]) -> Set[Region]:
+    ''' Check all region and merge adjacent ones '''
```

Comment:
This will not handle region fully included in other regions. It's not really supposed to happen but we could add a comment here?

Reply:
This is not my code, but I find the comment pretty spot on, it explains exactly what the code does. This is strictly for adjacent regions, nothing with overlap or inclusion.

(AFAIK we do expect them to be non-overlapping, this is just trying to avoid trivial fragmentation when two regions are directly next to each other and could have been one region from the start.)